oak/server/rust/oak_runtime/src/channel.rs

```diff
@@ -126,8 +147,23 @@ impl ChannelWriter {
         if self.is_orphan() {
             return Err(OakStatus::ERR_CHANNEL_CLOSED);
         }
-        let mut messages = self.messages.write().unwrap();
-        messages.push_back(msg);
+
+        {
+            let mut messages = self.messages.write().unwrap();
+
+            messages.push_back(msg);
+        }
+
+        let mut waiting_threads = self.waiting_threads.lock().unwrap();
+
+        // Unpark all the waiting threads, that still have references
```

Comment:
Should we remove the ones that no longer have references? Can they possibly come back?